contrib/grub/15_linux_zfs

```diff
@@ -638,38 +638,38 @@ zfs_linux_entry () {
     kernel="$7"
     kernel_additional_args="${8:-}"
 
-    kernel_version=$(basename ${kernel} | sed -e "s,^[^0-9]*-,,g")
-    submenu_indentation="$(printf %${submenu_level}s|tr " " "${grub_tab}")"
+    kernel_version=$(basename "${kernel}" | sed -e "s,^[^0-9]*-,,g")
+    submenu_indentation="$(printf %${submenu_level}s | tr " " "${grub_tab}")"
 
-    echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-${dataset}-${kernel_version}' {" | sed "s/^/$submenu_indentation/"
+    echo "menuentry '$(echo "${title}" | grub_quote)' ${CLASS} \${menuentry_id_option} 'gnulinux-${dataset}-${kernel_version}' {" | sed "s/^/${submenu_indentation}/"
 
-    if [ "$quick_boot" = 1 ]; then
-        echo "	recordfail" | sed "s/^/$submenu_indentation/"
+    if [ "${quick_boot}" = 1 ]; then
+        echo "	recordfail" | sed "s/^/${submenu_indentation}/"
     fi
 
     # Use ELILO's generic "efifb" when it's known to be available.
     # FIXME: We need an interface to select vesafb in case efifb can't be used.
-    if [ "x$GRUB_GFXPAYLOAD_LINUX" = x ]; then
-        echo "	load_video" | sed "s/^/$submenu_indentation/"
+    if [ "${GRUB_GFXPAYLOAD_LINUX}" = "" ]; then
+        echo "	load_video" | sed "s/^/${submenu_indentation}/"
     else
-        if [ "x$GRUB_GFXPAYLOAD_LINUX" != xtext ]; then
-        echo "	load_video" | sed "s/^/$submenu_indentation/"
+        if [ "${GRUB_GFXPAYLOAD_LINUX}" != xtext ]; then
+        echo "	load_video" | sed "s/^/${submenu_indentation}/"
         fi
     fi
 
-    if ([ "$ubuntu_recovery" = 0 ] || [ "${type}" != "recovery" ]) && \
-        ([ "x$GRUB_GFXPAYLOAD_LINUX" != x ] || [ "$gfxpayload_dynamic" = 1 ]); then
-        echo "	gfxmode \$linux_gfx_mode" | sed "s/^/$submenu_indentation/"
+    if ([ "${ubuntu_recovery}" = 0 ] || [ "${type}" != "recovery" ]) && \
+        ([ "${GRUB_GFXPAYLOAD_LINUX}" != "" ] || [ "${gfxpayload_dynamic}" = 1 ]); then
+        echo "	gfxmode \${linux_gfx_mode}" | sed "s/^/${submenu_indentation}/"
     fi
 
-    echo "	insmod gzio" | sed "s/^/$submenu_indentation/"
-    echo "	if [ x\$grub_platform = xxen ]; then insmod xzio; insmod lzopio; fi" | sed "s/^/$submenu_indentation/"
+    echo "	insmod gzio" | sed "s/^/${submenu_indentation}/"
+    echo "	if [ \${grub_platform} = xen ]; then insmod xzio; insmod lzopio; fi" | sed "s/^/${submenu_indentation}/"
```

Comment:
It'll break if for whatever reason $grub_platform is empty

Reply:
Thanks! Fixed in new commit.